### ISSUE TITLE
YOLO Predictor error

### ISSUE THREAD

**Issue by ahsan44411** [AlexeyAB/darknet#6721]
**Description of a bug**
I received this error after I trained my model.
This is the error
darknet: ./src/image.c:41: get_pixel: Assertion `x < m.w && y < m.h && c < m.c' failed.

It seems to me that the bounding boxes that YOLO is predicting have coordinates outside the image.

**Command I used**
!./darknet detector test data/obj.data cfg/yolov4-tiny.cfg /mydrive/text_detector_weights/yolov4-tiny_best.weights /content/1.jpeg -thresh 0.5

**System**
I was using COLAB which uses a Linux virtual environment.

**in what cases a bug occurs, and in which not?**
I get this error each time I run a prediction. Even if all the coordinats of the predicted bounding boxes are in the image I still get this error.

**Complete output of the command**

 CUDA-version: 10010 (10010), cuDNN: 7.6.5, CUDNN_HALF=1, GPU count: 1  
 CUDNN_HALF=1 
 OpenCV version: 3.2.0
 0 : compute_capability = 370, cudnn_half = 0, GPU: Tesla K80 
net.optimized_memory = 0 
mini_batch = 1, batch = 1, time_steps = 1, train = 0 
   layer   filters  size/strd(dil)      input                output
   0 conv     32       3 x 3/ 2    416 x 416 x   1 ->  208 x 208 x  32 0.025 BF
   1 conv     64       3 x 3/ 2    208 x 208 x  32 ->  104 x 104 x  64 0.399 BF
   2 conv     64       3 x 3/ 1    104 x 104 x  64 ->  104 x 104 x  64 0.797 BF
   3 route  2 		                       1/2 ->  104 x 104 x  32 
   4 conv     32       3 x 3/ 1    104 x 104 x  32 ->  104 x 104 x  32 0.199 BF
   5 conv     32       3 x 3/ 1    104 x 104 x  32 ->  104 x 104 x  32 0.199 BF
   6 route  5 4 	                           ->  104 x 104 x  64 
   7 conv     64       1 x 1/ 1    104 x 104 x  64 ->  104 x 104 x  64 0.089 BF
   8 route  2 7 	                           ->  104 x 104 x 128 
   9 max                2x 2/ 2    104 x 104 x 128 ->   52 x  52 x 128 0.001 BF
  10 conv    128       3 x 3/ 1     52 x  52 x 128 ->   52 x  52 x 128 0.797 BF
  11 route  10 		                       1/2 ->   52 x  52 x  64 
  12 conv     64       3 x 3/ 1     52 x  52 x  64 ->   52 x  52 x  64 0.199 BF
  13 conv     64       3 x 3/ 1     52 x  52 x  64 ->   52 x  52 x  64 0.199 BF
  14 route  13 12 	                           ->   52 x  52 x 128 
  15 conv    128       1 x 1/ 1     52 x  52 x 128 ->   52 x  52 x 128 0.089 BF
  16 route  10 15 	                           ->   52 x  52 x 256 
  17 max                2x 2/ 2     52 x  52 x 256 ->   26 x  26 x 256 0.001 BF
  18 conv    256       3 x 3/ 1     26 x  26 x 256 ->   26 x  26 x 256 0.797 BF
  19 route  18 		                       1/2 ->   26 x  26 x 128 
  20 conv    128       3 x 3/ 1     26 x  26 x 128 ->   26 x  26 x 128 0.199 BF
  21 conv    128       3 x 3/ 1     26 x  26 x 128 ->   26 x  26 x 128 0.199 BF
  22 route  21 20 	                           ->   26 x  26 x 256 
  23 conv    256       1 x 1/ 1     26 x  26 x 256 ->   26 x  26 x 256 0.089 BF
  24 route  18 23 	                           ->   26 x  26 x 512 
  25 max                2x 2/ 2     26 x  26 x 512 ->   13 x  13 x 512 0.000 BF
  26 conv    512       3 x 3/ 1     13 x  13 x 512 ->   13 x  13 x 512 0.797 BF
  27 conv    256       1 x 1/ 1     13 x  13 x 512 ->   13 x  13 x 256 0.044 BF
  28 conv    512       3 x 3/ 1     13 x  13 x 256 ->   13 x  13 x 512 0.399 BF
  29 conv     18       1 x 1/ 1     13 x  13 x 512 ->   13 x  13 x  18 0.003 BF
  30 yolo
[yolo] params: iou loss: ciou (4), iou_norm: 0.07, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.05
nms_kind: greedynms (1), beta = 0.600000 
  31 route  27 		                           ->   13 x  13 x 256 
  32 conv    128       1 x 1/ 1     13 x  13 x 256 ->   13 x  13 x 128 0.011 BF
  33 upsample                 2x    13 x  13 x 128 ->   26 x  26 x 128
  34 route  33 23 	                           ->   26 x  26 x 384 
  35 conv    256       3 x 3/ 1     26 x  26 x 384 ->   26 x  26 x 256 1.196 BF
  36 conv     18       1 x 1/ 1     26 x  26 x 256 ->   26 x  26 x  18 0.006 BF
  37 yolo
[yolo] params: iou loss: ciou (4), iou_norm: 0.07, obj_norm: 1.00, cls_norm: 1.00, delta_norm: 1.00, scale_x_y: 1.05
nms_kind: greedynms (1), beta = 0.600000 
Total BFLOPS 6.737 
avg_outputs = 299663 
 Allocate additional workspace_size = 12.46 MB 
Loading weights from /mydrive/text_detector_weights/yolov4-tiny_best.weights...
 seen 64, trained: 96 K-images (1 Kilo-batches_64) 
Done! Loaded 38 layers from weights-file 
 Detection layer: 30 - type = 28 
 Detection layer: 37 - type = 28 
/content/1.jpeg: Predicted in 14.931000 milli-seconds.
lines: 97%	(left_x:  157   top_y:  371   width:  289   height:   92)
lines: 100%	(left_x:  162   top_y:  556   width:  448   height:   77)
lines: 96%	(left_x:  163   top_y:  453   width:  421   height:   94)
lines: 100%	(left_x:  812   top_y:  629   width:  409   height:   53)
lines: 100%	(left_x:  846   top_y:  570   width:  644   height:   54)
lines: 94%	(left_x:  861   top_y:  289   width:  491   height:   57)
lines: 100%	(left_x:  884   top_y:  453   width:  469   height:   64)
lines: 100%	(left_x:  892   top_y:  378   width:  512   height:   61)
darknet: ./src/image.c:41: get_pixel: Assertion `x < m.w && y < m.h && c < m.c' failed.

**Comment by stephanecharette** [comment]
> It seems to me that the bounding boxes that YOLO is predicting have coordinates outside the image.

Correct, the coordinates returned by Darknet will sometimes be beyond the edges of the image.

If it is useful to you, DarkHelp has an option that truncates and limits the coordinates to the size of the image:  https://www.ccoderun.ca/darkhelp/api/classDarkHelp.html#ab3556e5fc64a38665b87d72df0746e6a

This option is 'on' by default, but can be turned off to get the exact same results as Darknet.

**Comment by XueFengHuang666** [comment]
I got the same error as you, for me , it works well while training on RGB images and testing on RGB images, but images of my project are single channel,  so I just change the channel=1 on the .cfg file, training works well but error occurs while testing on images(also single channel). Have you solved the problem?  Great appreciate your answer!
